services/controler.js

```diff
@@ -1,11 +1,86 @@
-import logger from '../services/logger.js'
-
-const images = (req,res)=>{
-    let reqURI = req.originalUrl;
-    
-    logger.debug('reqURI: ' + reqURI)
-    
-    res.status(200).json({message:  `You requested the following URI: ${reqURI}`})
+import establishRedis from '../services/redis.js';
+import logger from '../services/logger.js';
+
+// List of available target sizes
+const targetSizes = [32, 64, 128, 256, 512, 1024, 2048, 4096, 8192];
+
+// Find the nearest available resolution given the requested size
+function getNearestResolution(requestedSize) {
+    const reqNum = parseInt(requestedSize, 10);
+    if (isNaN(reqNum)) return 'original';
+    const nearest = targetSizes.reduce((prev, curr) =>
+        Math.abs(curr - reqNum) < Math.abs(prev - reqNum) ? curr : prev
+    );
+    return `${nearest}x${nearest}`;
```

Comment:
No need to iterate through the entire array using an (IMO) unreadable reduce when we can just return after the first fitting size.
```suggestion
    for (const current of targetSize) {
        if (current >= reqNum) return `${current}x${current}`
    }
    return 'original';
```